### DISCUSSION TITLE
CDC high speed in full speed mode

### DISCUSSION THREAD

**Discussion by emonjak** [hathach/tinyusb#1600] [Q&A]
Hi all,
we use tinyusb on STM32H743. USB is configured to high speed, but runs in full speed mode because of USB 1.1 hub.  It looks like the max. packet size for CDC transmitt is still 512 bytes. Therefore, the sending of the strings with the 64 characters length is not completed with ZLP. Call 
tud_cdc_write(NULL, 0);
tud_cdc_write_flush();
has no effect.
Sending a string with 512 characters works correctly.

Any ideas how to trigger the sending of a ZLP from CDC?
I know about usbd_edpt_xfer() but as shown by 512 character string it is no need for CDC.

**Comment by hathach** [comment]
right, I think this is a bug in cdc driver, it should use actual speed rather than the max speed. I have made an issue based on this as reminder to fix this in the future.
